pkg/utils/chart/chart.go

```diff
@@ -77,6 +84,20 @@ func (c *Chart) Apply(
 	if err != nil {
 		return errors.Wrapf(err, "could not apply chart '%s' in namespace '%s'", c.Name, namespace)
 	}
+
+	// Delete disabled subcharts
+	for _, sc := range c.SubCharts {
+		v, err := evaluateCondition(additionalValues, sc.Condition)
+		if err != nil {
+			return errors.Wrapf(err, "could not evaluate condition %s for subchart %s", sc.Condition, sc.Name)
+		}
+		if !v {
+			if err := sc.Delete(ctx, client, namespace); err != nil {
```

Comment:
How 'not found' and 'no match' errors are being handled?

Reply:
You can see the `Object.Delete` code a few lines below: https://github.com/gardener/gardener/blob/d5fd42721faa352ffbed071c0314d25f158b472a/pkg/utils/chart/chart.go#L159-L162, a `nil` is returned on `NotFound`. `Chart.Delete` resolves recursively to a series of `Object.Delete` invocations. No need to handle such errors here or elsewhere in this file where `Chart.Delete` is used.

Reply:
I noticed that `client.Delete` a few lines below the above doesn't handle `NotFound` and added it just in case.